src/edu/stanford/nlp/optimization/GoldenSectionLineSearch.java

```diff
@@ -161,10 +161,8 @@ public double minimize(DoubleUnaryOperator function) {
    * dump the {@code <x,y>} pairs it computed found
    */
   public void dumpMemory() {
-    Double[] keys = memory.keySet().toArray(new Double[memory.keySet().size()]);
-    Arrays.sort(keys);
-    for (Double key : keys) {
-      log.info(key + "\t" + memory.get(key));
+    for (Map.Entry<Double, Double> entry : new TreeMap<>(memory).entrySet()) {
```

Comment:
Here is another use of sorting via TreeMap.  It's just a debug method, but I don't like the functionality change